db/blip_sync_context.go

```diff
@@ -677,6 +698,42 @@ func toHistory(revisions Revisions, knownRevs map[string]bool, maxHistory int) [
 	return history
 }
 
+// toHistoryForHLV formats blip History property for V4 replication and above
+func toHistoryForHLV(hlv *HybridLogicalVector) string {
```

Comment:
This definitely feels like it should be a method on HLV.  And similar to the case above, this will be a common enough operation that we want to optimize it and preferably store it in this format in the HLV.  (i.e. if there are 10K users all replicating the same revision from the rev cache, we don't want to have to compute this uniquely for each user)